libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -120,6 +122,52 @@ void AP_DDS_Client::update_topic(sensor_msgs_msg_NavSatFix& msg, const uint8_t i
     msg.position_covariance[8] = vdopSq;
 }
 
+#include "generated/TransformStamped.h"
+
+void AP_DDS_Client::populate_static_transforms(tf2_msgs_msg_TFMessage& msg)
+{
+    msg.transforms_size = 0;
+
+    auto &gps = AP::gps();
+    for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
+        const auto gps_type = gps.get_type(i);
+        if (gps_type == AP_GPS::GPS_Type::GPS_TYPE_NONE) {
+            continue;
+        }
+        update_topic(msg.transforms[i].header.stamp);
+        char gps_frame_id[16];
+        //! @todo should GPS frame ID's be 0 or 1 indexed in ROS?
+        snprintf(gps_frame_id, 16, "GPS_%u", i);
```

Comment:
best to use sizeof() for the size in snprintf

Reply:
I went down the rabbit hole a bit. 
https://stackoverflow.com/a/35401865


Fixed it in aab9fe921b8d5af97c21d4b6a1aedfdea4af14b6 with a `sizeof`